cmd/waitlist/handlers.go

```diff
@@ -0,0 +1,141 @@
+package main
+
+import (
+	"github.com/go-openapi/runtime/middleware"
+	uuid "github.com/satori/go.uuid"
+
+	"github.com/iryonetwork/wwm/gen/waitlist/restapi/operations/item"
+	"github.com/iryonetwork/wwm/gen/waitlist/restapi/operations/waitlist"
+	storage "github.com/iryonetwork/wwm/storage/waitlist"
+	"github.com/iryonetwork/wwm/utils"
+)
+
+type handlers struct {
+	s storage.Storage
+}
+
+func (h *handlers) WaitlistGet() waitlist.GetHandler {
+	return waitlist.GetHandlerFunc(func(params waitlist.GetParams, principal *string) middleware.Responder {
+		lists, err := h.s.Lists()
+		if err != nil {
+			return utils.NewErrorResponse(err)
+		}
+
+		return waitlist.NewGetOK().WithPayload(lists)
+	})
+}
+
+func (h *handlers) WaitlistPost() waitlist.PostHandler {
+	return waitlist.PostHandlerFunc(func(params waitlist.PostParams, principal *string) middleware.Responder {
+		list, err := h.s.AddList(*params.List.Name)
+		if err != nil {
+			return utils.NewErrorResponse(err)
+		}
+
+		return waitlist.NewPostCreated().WithPayload(list)
+	})
+}
+
+func (h *handlers) WaitlistPutListID() waitlist.PutListIDHandler {
+	return waitlist.PutListIDHandlerFunc(func(params waitlist.PutListIDParams, principal *string) middleware.Responder {
+		if params.ListID != params.List.ID {
+			return utils.NewError(utils.ErrBadRequest, "URL list ID and body list ID do not match")
+		}
+
+		_, err := h.s.UpdateList(params.List)
+		if err != nil {
+			return utils.NewErrorResponse(err)
+		}
+
+		return waitlist.NewPutListIDNoContent()
+	})
+}
+
+func (h *handlers) WaitlistDeleteListID() waitlist.DeleteListIDHandler {
+	return waitlist.DeleteListIDHandlerFunc(func(params waitlist.DeleteListIDParams, principal *string) middleware.Responder {
+		id, err := uuid.FromString(params.ListID)
```

Comment:
You can mark the ID as `uuid` with `format: uuid`

Reply:
The ListID will then already checked and passed as type UUID

Reply:
That's great, used `format: uuid` for all uuids.